src/core/net/udp6.cpp

```diff
@@ -372,7 +372,7 @@ otError Udp::SendTo(SocketHandle &aSocket, Message &aMessage, const MessageInfo
     messageInfoLocal.SetSockPort(aSocket.GetSockName().mPort);
 
 #if OPENTHREAD_CONFIG_PLATFORM_UDP_ENABLE
-    if (ShouldUsePlatformUdp(aSocket))
+    if (ShouldUsePlatformUdp(aSocket) && messageInfoLocal.IsHostInterface())
```

Comment:
Messages that are sent to Thread devices are also passed to Host tun interface in PLAT_UDP implementation.
So I think we should not check for `.IsHostInterface())` in this place.

We have two kinds of UDP sockets:
1. UDP sockets not using Platform UDP (e.g. MLE, TMF)
  - For these UDP sockets, IsHostInterface() is generally set to false for sending and received UDP messages. 
1. UDP sockets using Platform UDP
  - For these UDP sockets, IsHostInterface() tells where should the message be sent and from which host interface the received message is from. For example, `IsHostInterface` is set to false if the receiving host interface is `wpan0`, and is set to true otherwise.